### PR TITLE
Inline Help: Admin-Manage page

### PR DESCRIPTION
**Pull request by tk21**
dependent files: introjs.cs, intro-injector.js, intro.js... added to 'base_manage' and 'base'

intro-injector.js finished, please log in as admin and view "manage" page for for preview by clicking on the "show me how" button (button placement is temporary... I will figure out placement/design for that later)

I also need to figure out why the numbers with the steps are positioned oddly/cropped.. will iterate on that next week.

### CODE REVIEW

**Comment by aronasorman** [comment]
Please follow the guidelines in https://github.com/learningequality/ka-lite/blob/develop/CONTRIBUTING.md when writing PR descriptions.

**Comment by rtibbles** [comment]
You may also want to rebase your branch to latest develop for ease of merging!

**Comment by MCGallaspy** [comment]
You don't have to open a new PR, you can simply continue to push to your own branch and add/edit comments here.

File: kalite/distributed/static/js/distributed/intro-injector.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
+//on page load
+$(function() {
+  console.log( "ready!" );
```
Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/intro-injector.js]: Eventually you'll want to get rid of this, or if you feel that console output could be helpful then make a more meaningful message.

File: kalite/distributed/static/js/distributed/intro-injector.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  //Use relative URL to determine which YAML file to parse
```
Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/intro-injector.js]: I know I helped you write the comments, but you should change them to reflect what's actually going on. Namely there's no YAML around these parts -- that should all be handled on the server-side.

File: kalite/distributed/static/js/distributed/intro-injector.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  var relative_url = window.location.pathname;
+  var response = getIntro(relative_url);
+  
+  //Parse YAML file into JSON to be used with intro.js !!! this is actually just a json object for nao.
```
Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/intro-injector.js]: The comment is inaccurate.

File: kalite/distributed/static/js/distributed/intro-injector.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+console.log(jsonResponse);
+
+  //Load from our new JSON object, and attach to elements in page
+  attachAttributes(jsonResponse);
+
+  console.log("uhh hello?");
+
+});
+
+function startTour() {
```
Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/intro-injector.js]: You should add explanatory comments about functions. We don't seem to have a set-in-stone style for JS yet, but at a minimum I would say indicate what the expected parameters are (in this case none), what the intended use of the function is and its side effects, and what the function returns.

File: kalite/distributed/static/js/distributed/intro-injector.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+//make an ajax request to some url that we haven' tyet defined and then we would wait until
+//the server returns a response and what we expet is a json object (Serialized)
+//that we can parse directly into an actual json object
+//
+function getIntro(relative_url) {
+  return {"management/zone/None/": [{"a#manage.admin-only": [{"step": 1}, {"text": "Welcome! This is the landing page for admins. If at any point you would like to navigate back to this page, click on this tab!"}]}, {"li.facility": [{"step": 2}, {"text": "Clicking on this tab will show you a quick overview of all facilities and devices"}]}, {"a.create-facility": [{"step": 3}, {"text": "To add new facilities, click here..."}]}, {"content-container .row:nth-child(3)": [{"step": 4}, {"text": "Information on your device status will be shown here"}]}, {"a.create-facility": [{"step": 5}, {"text": "If you decide to register your device to our central club, click here"}]}, {"#not-registered": [{"step": 6}, {"text": "Questions about anything on the page? Be sure to consult the user manual or FAQ for more detailed walk throughs!"}]}]};
+}
+
+//Given JSON form of YAML file for this page, parse and return JSON object
```
Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/intro-injector.js]: Inaccurate comment, or at least a bit vague.

File: kalite/distributed/static/js/distributed/intro-injector.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+      console.log("THIS IS SELECTOR KEY AT THE END THO:" +selectorKey);
+    });
+  }); 
+
+
+/**
+  var selectorKey = "a#manage.admin-only";
+  var key = "data-step";
+  var value = 1;
+  $(selectorKey).attr(key, value); **/
```
Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/intro-injector.js]: Don't leave dead code floating around... it's spooky!

File: kalite/distributed/templates/distributed/base_manage.html [hunk 7]
```diff
@@ -25,4 +26,13 @@
...
-{% endblock subnavbar %}
\ No newline at end of file
+{% endblock subnavbar %}
+    <script>
+        function startTour() {
+            var tour = introJs()
+            tour.setOption('tooltipPosition', 'auto');
+            tour.setOption('positionPrecedence', ['left', 'right', 'bottom', 'top'])
+            tour.start()
+        }
```
Comment (MCGallaspy) [inline kalite/distributed/templates/distributed/base_manage.html]: Beware! Inline script is a tool of great evil. :P

**Comment by MCGallaspy** [comment]
Stale. Reopen when it's ready.

**Pull request closed without merging**